networkx/drawing/tests/test_latex.py

```diff
@@ -0,0 +1,289 @@
+import filecmp
+import os
+
+import pytest
+
+import networkx as nx
+
+
+def test_tikz_attributes():
+    G = nx.path_graph(4, create_using=nx.DiGraph)
+    G.add_edge(0, 0)
+    pos = {n: (n, n) for n in G}
+
+    G.nodes[0]["style"] = "blue"
+    G.nodes[2]["style"] = "line width 3,blue!50"
+    G.nodes[3]["label"] = "Stop"
+    G.edges[(1, 2)]["style"] = "line width 3,green!90"
+    G.edges[(2, 3)]["style"] = "green"
+    G.edges[(0, 1)]["label"] = "1st Step"
+    G.edges[(0, 1)]["label_options"] = "near start"
+    G.edges[(2, 3)]["label"] = "3rd Step"
+    G.edges[(2, 3)]["label_options"] = "near end"
+    G.edges[(1, 2)]["label"] = "2nd"
+
+    output_tex = nx.to_latex(
+        G,
+        pos=pos,
+        as_document=False,
+        tikz_options="[scale=3]",
+        node_options="style",
+        edge_options="style",
+        node_label="label",
+        edge_label="label",
+        edge_label_options="label_options",
+    )
+    expected_tex = r"""\begin{figure}
+  \begin{tikzpicture}[scale=3]
+      \draw
+        (0, 0) node[blue] (0){0}
+        (1, 1) node (1){1}
+        (2, 2) node[line width 3,blue!50] (2){2}
+        (3, 3) node (3){Stop};
+      \begin{scope}[->]
+        \draw (0) to (1) node[near start] {1st Step};
+        \draw (0) to[loop,] (0);
+        \draw (1) to[line width 3,green!90] (2) node[] {2nd};
+        \draw (2) to[green] (3) node[near end] {3rd Step};
+      \end{scope}
+    \end{tikzpicture}
+\end{figure}
+"""
+    print(output_tex)
+    # Pretty way to assert that A.to_document() == expected_tex
+    content_same = True
+    for aa, bb in zip(expected_tex.split("\n"), output_tex.split("\n")):
+        if aa != bb:
+            content_same = False
+            print(f"-{aa}|\n+{bb}|")
```

Comment:
This entire block of code is only there to print the parts that are different so that when pytest runs you can see the difference in the results compared to expected. We should either keep it all or remove it all (maybe best to leave it there commented out?  Commented out code eventually gets out of date. But what about code that is helpful for people who are debugging failed tests?  There isn't really a place for that in the normal workflow.).  :}